rust/tools/authorization-logic/src/ast.rs

```diff
@@ -72,10 +79,26 @@ pub enum AstFact {
     AstCanSayFact { p: AstPrincipal, f: Box<AstFact> },
 }
 
+#[derive(Copy, Debug, Clone, Serialize, Deserialize)]
+pub enum AstOperator {
+    LessThan,
+    GreaterThan,
+    Equals,
+    NotEquals,
+    LessOrEquals,
+    GreaterOrEquals
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum AstRValue {
```

Comment:
What is an `RValue`?

Reply:
Here a `RValue`s are the parts of the grammar that can appear on the RHS. The names `rvalue` and `lvalue` are commonly used in programming language designs to describe parts of grammars that can appear on the right hand side and left hand side of assignment statements respectively. So for example in `x := y + 7` the rvalue is `x` and the lvalue is `y + 7`. [C++ is one example of a language that used to use this terminology -- the grammar has gotten more complicated though apparently](https://stackoverflow.com/questions/9406121/what-exactly-is-an-r-value-in-c#:~:text=In%20C%2B%2B%2C%20an%20rvalue,which%20is%20not%20a%20reference.&text=In%20(1)%2C%20the%20temporary,string%20literal%20is%20an%20rvalue.) It's not just C++ that uses this terminology in its language design, though.

So here the name rvalue is used to describe the part of the grammar that can appear on the RHS of a conditional horn clause, which is not exactly the same as appearing on the RHS of an assignment, but I thought it would be intuitive for people familiar with the term from other language designs. I'm open to other names.

Reply:
I am aware of `rvalue` and  I think it fits the usage here. Can you just document it briefly thouugh?